moveit_plugins/moveit_simple_controller_manager/src/follow_joint_trajectory_controller_handle.cpp

```diff
@@ -36,50 +36,61 @@
 /* Author: Michael Ferguson, Ioan Sucan, E. Gil Jones */
 
 #include <moveit_simple_controller_manager/follow_joint_trajectory_controller_handle.h>
-#include <moveit/utils/xmlrpc_casts.h>
+// #include <moveit/utils/xmlrpc_casts.h>
 
-using namespace moveit::core;
-static const std::string LOGNAME("SimpleControllerManager");
+// using namespace moveit::core;
+
+using namespace std::placeholders;
 
 namespace moveit_simple_controller_manager
 {
 bool FollowJointTrajectoryControllerHandle::sendTrajectory(const moveit_msgs::msg::RobotTrajectory& trajectory)
 {
-  ROS_DEBUG_STREAM_NAMED(LOGNAME, "new trajectory to " << name_);
+  RCLCPP_DEBUG_STREAM(LOGGER, "new trajectory to " << name_);
 
   if (!controller_action_client_)
     return false;
 
   if (!trajectory.multi_dof_joint_trajectory.points.empty())
   {
-    ROS_WARN_NAMED(LOGNAME, "%s cannot execute multi-dof trajectories.", name_.c_str());
+    RCLCPP_WARN_STREAM(LOGGER, name_ << " cannot execute multi-dof trajectories.");
   }
 
   if (done_)
-    ROS_DEBUG_STREAM_NAMED(LOGNAME, "sending trajectory to " << name_);
+    RCLCPP_DEBUG_STREAM(LOGGER, "sending trajectory to " << name_);
   else
-    ROS_DEBUG_STREAM_NAMED(LOGNAME, "sending continuation for the currently executed trajectory to " << name_);
+    RCLCPP_DEBUG_STREAM(LOGGER, "sending continuation for the currently executed trajectory to " << name_);
 
-  control_msgs::FollowJointTrajectoryGoal goal = goal_template_;
+  control_msgs::action::FollowJointTrajectory::Goal goal = goal_template_;
   goal.trajectory = trajectory.joint_trajectory;
-  controller_action_client_->sendGoal(
-      goal, boost::bind(&FollowJointTrajectoryControllerHandle::controllerDoneCallback, this, _1, _2),
-      boost::bind(&FollowJointTrajectoryControllerHandle::controllerActiveCallback, this),
-      boost::bind(&FollowJointTrajectoryControllerHandle::controllerFeedbackCallback, this, _1));
+
+  rclcpp_action::Client<control_msgs::action::FollowJointTrajectory>::SendGoalOptions send_goal_options;
+  // Active callback
+  send_goal_options.goal_response_callback = [this](const auto& /* unused-arg */) {
+    RCLCPP_DEBUG_STREAM(LOGGER, name_ << " started execution");
+  };
+  // Result callback
+  send_goal_options.result_callback =
+      std::bind(&FollowJointTrajectoryControllerHandle::controllerDoneCallback, this, _1);
+  // Send goal
+  controller_action_client_->async_send_goal(goal, send_goal_options);
+  // TODO(JafarAbdi): current_goal_ .?
   done_ = false;
   last_exec_ = moveit_controller_manager::ExecutionStatus::RUNNING;
   return true;
 }
 
-void FollowJointTrajectoryControllerHandle::configure(XmlRpc::XmlRpcValue& config)
```

Comment:
Is this function needed .? path_tolerance, goal_tolerance, and goal_time_tolerance aren't set anywhere !